### PR TITLE
Add counter_user_followers and changes to migrations

### PR DESCRIPTION
**Pull request by tckthecreator**
## What this PR introduces?

Add CounterUserFollowers and changes userId BIGSERIAL -> userId VARCHAR(36)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] I followed GitFlow pattern to create the branch
- [x] My code produces no warnings or errors in the console
- [x] Tests for the changes have been added / updated
- [x] Mocks fot the database have been added / updated

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Documentation update
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] CI/CD related changes
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/database/migrations/1639493810363-Create-CounterGameUsers-Table.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@
 const name = "1639493810363-Create-CounterGameUsers-Table";
 
 const up = `CREATE TABLE IF NOT EXISTS counter_game_users(
-	gameId BIGINT NOT NULL,
+	gameId BIGSERIAL NOT NULL,
```
Comment (henriqueleite42) [inline src/database/migrations/1639493810363-Create-CounterGameUsers-Table.ts:11]: Revert change

File: src/database/migrations/1639498893009-Create-Reports-Table.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@
 const name = "1639498893009-Create-Reports-Table";
 
 const up = `CREATE TABLE IF NOT EXISTS reports (
-	id BIGSERIAL NOT NULL,
-	userId BIGINT NOT NULL,
+	id VARCHAR(36) NOT NULL,
```
Comment (henriqueleite42) [inline src/database/migrations/1639498893009-Create-Reports-Table.ts:11]: Revert change

File: src/database/migrations/1640178273626-Create-CounterUserFollowers.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	FOREIGN KEY (userId) REFERENCES users (id) ON DELETE CASCADE
+);
+
+CREATE OR REPLACE FUNCTION update_counter_user_followers() RETURNS TRIGGER AS
+$$
+BEGIN
+	IF TG_OP = 'INSERT' THEN
+		INSERT INTO counter_user_followers (userId, followers)
+		VALUES (new.userId, 1)
```
Comment (henriqueleite42) [inline src/database/migrations/1640178273626-Create-CounterUserFollowers.ts:24]: To count the user followers u should use the `followedId` column.
The `userId` should be used in the task:
https://captisapp.atlassian.net/browse/CA-349

**Review by henriqueleite42 — CHANGES_REQUESTED** [review]

File: src/database/migrations/1639498893009-Create-Reports-Table.ts (hunk 2 above)
Comment (tckthecreator) [reply src/database/migrations/1639498893009-Create-Reports-Table.ts:11]: both or only the id?

Comment (tckthecreator) [reply src/database/migrations/1639498893009-Create-Reports-Table.ts:11]: > Revert change

both or only the id?

File: src/database/migrations/1640178273626-Create-CounterUserFollowers-Table.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	FOREIGN KEY (userId) REFERENCES users (id) ON DELETE CASCADE
+);
+
+CREATE OR REPLACE FUNCTION update_counter_user_followers() RETURNS TRIGGER AS
+$$
+BEGIN
+	IF TG_OP = 'INSERT' THEN
+		INSERT INTO counter_user_followers (userId, followers)
+		VALUES (new.userId, 1)
```
Comment (henriqueleite42) [inline src/database/migrations/1640178273626-Create-CounterUserFollowers-Table.ts:24]: Sorry, my bad, this really should be `followedId`.
As we r counting the **followers of someone**, it is related to the `followedId` (people who is being followed).

**Review by henriqueleite42 — CHANGES_REQUESTED** [review]

**Review by henriqueleite42 — APPROVED** [review]

**Pull request merged**